### PR TITLE
subversion: 1.12.2 -> 1.14.1

### PR DESCRIPTION
**Pull request by ajs124**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/pull/120562#issuecomment-826599361

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by dotlambda — CHANGES_REQUESTED** [review]
fails to build for me

**Comment by mkg20001** [comment]
@dotlambda Local build works for me - can I have your logs?

If you have sandbox disabled, this could be it. Do your builds use the sandbox?

**Comment by dotlambda** [comment]
Oh sorry, I didn't specify it's `subversionClient` that's failing.

**Comment by mkg20001** [comment]
@ajs124 The subversionClient is just an overriden version of the subversion package.

The error is
```
SWIG python disabled at configure time: py3c library not found
```

would need to be added to pythonpackages set propably.

**Comment by ajs124** [comment]
Oops. I though I'd tested that. Should be fixed now. Thanks for catching that.

**Comment by ajs124** [comment]
> pythonpackages set propably

I added it as a regular package instead, because it's just a bunch of C headers, but if you install it with buildPythonPackage/setup.py, it's installed as a python library and missing the pkg-config file.

File: pkgs/development/libraries/py3c/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    owner = "encukou";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "04i2z7hrig78clc59q3i1z2hh24g7z1bfvxznlzxv00d4s57nhpi";
+  };
+
+  makeFlags = [
+    "prefix=${placeholder "out"}"
+  ];
+
```
Comment (dotlambda) [inline pkgs/development/libraries/py3c/default.nix:17]:
```suggestion

  doCheck = true;
  
  checkInputs = [
    python2
    python3
  ];
```

**Comment by d1egoaz** [comment]
👋 Hello @ajs124 

I _think_ this broke my build, I'm on MacOS version 11.3 (20E232).
I'm new to Nix, so forgive me if the following doesn't make sense.

TBH I'm not sure what package is depending on this change, I'm using home-manager, when I'm trying to build my packages from the `nixos-unstable-small` branch I'm getting the following error:

```
clang-7: error: argument unused during compilation: '-fno-strict-overflow' [-Werror,-Wunused-command-line-argument]
error: command 'clang' failed with exi
```

Details:
<details><summary>Error</summary>
<p>


```
build flags: SHELL=/nix/store/30njb8l701pwnm5ya749fh2cgyc2d70m-bash-4.4-p23/bin/bash prefix=/nix/store/mx576n9ivgd25l1ay5v52pnva1ahj2wy-py3c-1.3.1
echo "Available targets:"
Available targets:
echo "    make test"
    make test
echo "    make test-<interpreter>"
    make test-<interpreter>
echo "    make doc"
    make doc
echo "    make install"
    make install
echo "    make clean"
    make clean
running tests
check flags: SHELL=/nix/store/30njb8l701pwnm5ya749fh2cgyc2d70m-bash-4.4-p23/bin/bash prefix=/nix/store/mx576n9ivgd25l1ay5v52pnva1ahj2wy-py3c-1.3.1 VERBOSE=y test
make test-python2
make[1]: Entering directory '/private/tmp/nix-build-py3c-1.3.1.drv-0/source'
cd test; rm -rvf build ; python2 setup.py build
running build
running build_ext
building 'test_py3c' extension
creating build
creating build/temp.macosx-10.12-x86_64-2.7
clang -I../include -I/nix/store/25lzm81cpg9jqasc9pccl9dzx9gh7rdl-python-2.7.18/include/python2.7 -c test_py3c.c -o build/temp.macosx-10.12-x86_64-2.7/test_py3c.o -Werror -Wall
creating build/lib.macosx-10.12-x86_64-2.7
clang -bundle -undefined dynamic_lookup build/temp.macosx-10.12-x86_64-2.7/test_py3c.o -L/nix/store/25lzm81cpg9jqasc9pccl9dzx9gh7rdl-python-2.7.18/lib -o build/lib.macosx-10.12-x86_64-2.7/test_py3c.so
PYTHONPATH=test/build/lib.macosx-10.12-x86_64-2.7 python2 test -v
test_get_context (__main__.CapsuleChecks) ... ok
test_get_destructor (__main__.CapsuleChecks) ... ok
test_get_name (__main__.CapsuleChecks) ... ok
test_get_pointer (__main__.CapsuleChecks) ... ok
test_get_pointer_nullname (__main__.CapsuleChecks) ... ok
test_import (__main__.CapsuleChecks) ... ok
test_invalid (__main__.CapsuleChecks) ... ok
test_invalid_operations (__main__.CapsuleChecks) ... ok
test_set_context (__main__.CapsuleChecks) ... ok
test_setdestructor (__main__.CapsuleChecks) ... ok
test_setname (__main__.CapsuleChecks) ... ok
test_setpointer (__main__.CapsuleChecks) ... ok
test_type (__main__.CapsuleChecks) ... ok
test_valid (__main__.CapsuleChecks) ... ok
test_return_notimplemented (__main__.ComparisonHelperChecks) ... ok
test_unreachable (__main__.ComparisonHelperChecks) ... ok
test_bad_type (__main__.FileShimChecks) ... ok
test_read (__main__.FileShimChecks) ... ok
test_read_write (__main__.FileShimChecks) ... ok
test_write (__main__.FileShimChecks) ... ok
test_fromstring (__main__.FloatChecks) ... ok
test_as (__main__.IntChecks) ... ok
test_check (__main__.IntChecks) ... ok
test_check_exact (__main__.IntChecks) ... ok
test_from (__main__.IntChecks) ... ok
test_type (__main__.IntChecks) ... ok
test_all_comparisons (__main__.NewComparisonChecks) ... ok
test_number (__main__.NewComparisonChecks) ... ok
test_number_int_equality (__main__.NewComparisonChecks) ... ok
test_number_number_equality (__main__.NewComparisonChecks) ... ok
test_all_comparisons (__main__.OldComparisonChecks) ... ok
test_number (__main__.OldComparisonChecks) ... ok
test_number_int_equality (__main__.OldComparisonChecks) ... ok
test_number_number_equality (__main__.OldComparisonChecks) ... ok
test_raw_calloc_zero (__main__.RawMallocChecks) ... ok
test_raw_malloc (__main__.RawMallocChecks) ... ok
test_raw_malloc_zerosize (__main__.RawMallocChecks) ... ok
test_raw_realloc (__main__.RawMallocChecks) ... ok
test_raw_realloc_null (__main__.RawMallocChecks) ... ok
test_raw_realloc_zero (__main__.RawMallocChecks) ... ok
test_asstring (__main__.StringChecks) ... ok
test_asutf8 (__main__.StringChecks) ... ok
test_asutf8andsize (__main__.StringChecks) ... ok
test_asutf8string (__main__.StringChecks) ... ok
test_check (__main__.StringChecks) ... ok
test_check_exact (__main__.StringChecks) ... ok
test_concat (__main__.StringChecks) ... ok
test_decode (__main__.StringChecks) ... ok
test_format (__main__.StringChecks) ... ok
test_fromformat (__main__.StringChecks) ... ok
test_fromstring (__main__.StringChecks) ... ok
test_fromstringandsize (__main__.StringChecks) ... ok
test_internfromstring (__main__.StringChecks) ... ok
test_interninplace (__main__.StringChecks) ... ok
test_type (__main__.StringChecks) ... ok

----------------------------------------------------------------------
Ran 55 tests in 0.073s

OK
make[1]: Leaving directory '/private/tmp/nix-build-py3c-1.3.1.drv-0/source'
make test-python3
make[1]: Entering directory '/private/tmp/nix-build-py3c-1.3.1.drv-0/source'
cd test; rm -rvf build ; python3 setup.py build
removed 'build/temp.macosx-10.12-x86_64-2.7/test_py3c.o'
removed directory 'build/temp.macosx-10.12-x86_64-2.7'
removed 'build/lib.macosx-10.12-x86_64-2.7/test_py3c.so'
removed directory 'build/lib.macosx-10.12-x86_64-2.7'
removed directory 'build'
running build
running build_ext
building 'test_py3c' extension
creating build
creating build/temp.macosx-10.6-x86_64-3.8
clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -I../include -I/nix/store/l4q9nfrq3wz4qgk5ikbzfcs7iqdb32y1-python3-3.8.9/include/python3.8 -c test_py3c.c -o build/temp.macosx-10.6-x86_64-3.8/test_py3c.o -Werror -Wall
clang-7: error: argument unused during compilation: '-fno-strict-overflow' [-Werror,-Wunused-command-line-argument]
error: command 'clang' failed with exit status 1
make[1]: *** [Makefile:27: build-python3] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-py3c-1.3.1.drv-0/source'
make: *** [Makefile:20: test] Error 2
builder for '/nix/store/9ibmpqmyykj626i34ggjdbfwhcb4iikq-py3c-1.3.1.drv' failed with exit code 2
cannot build derivation '/nix/store/s4ikp3xqcg9v2agr5dblw8aa0hwv8r86-subversion-1.14.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/67rys6v5p67cgb9h4hv6mgb7sf4aphqk-git-2.31.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/zag8im9ix51fnjvycjzc628shj9zln8n-home-manager-fonts.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/hs6lc5sgyxslhysqh52j63n2z4j8za3q-home-manager-path.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/yblwlblbja33z33als1qv1fgwypprbmz-home-manager-generation.drv': 1 dependencies couldn't be built
error: build of '/nix/store/yblwlblbja33z33als1qv1fgwypprbmz-home-manager-generation.drv' failed
```


</p>
</details>

If I use the commit revision just before this PR (5ef56ad9f17993bda954e5c0527984b3d6fa570a), things works fine.

Thanks

**Comment by dotlambda** [comment]
I can reproduce the error on Linux using `nix-build -E "with import ./. { }; py3c.override { stdenv = clangStdenv; }"`.

**Pull request merged**